### PR TITLE
1k user mesage for sirbot

### PR DESCRIPTION
**Pull request by nik849**
@ovv, I have been trying to test this with ngrok - I haven't got as far as I would like - hopefully you can see what I am trying to do here - put a listener for the event and a function to get the members in general and % 1000 etc ...

### CODE REVIEW

File: sirbot/pythondev/slack.py [hunk 1]
```diff
@@ -470,3 +472,13 @@ def add(self, slack):
                             " `{0}`".format(command.text)
 
         await slack.send(response)
+
+        async def members_joined(self, event, slack, _):
```
Comment (ovv) [inline sirbot/pythondev/slack.py]: you don't need to `await slack.channels` multiple time. It returns a channel object.

The most efficient would be to request it one time to check if the event is for #general and then a second time with the `update` argument to be sure we update the total members count. You can then use the object for the `to` and counting the number of members

File: sirbot/pythondev/slack.py [hunk 2]
```diff
@@ -470,3 +472,13 @@ def add(self, slack):
...
+            if event['channel'] == await slack.channels.get(name='general').id:
+                await slack.channels.get('general', update=True)
+                members = await slack.channels.get(name='general').members
+                if (len(members) % 1000) == 0:
+                    to = await slack.channels.get(name='general').members
+                    message = SlackMessage(to=to)
+                    message.text = 'We have just reached {} members!'.format()
```
Comment (ovv) [inline sirbot/pythondev/slack.py]: you forgot to put something in the `format`. I guess it should be the `len(members)`

File: sirbot/pythondev/slack.py [hunk 3]
```diff
@@ -470,3 +472,13 @@ def add(self, slack):
                             " `{0}`".format(command.text)
 
         await slack.send(response)
+
+        async def members_joined(self, event, slack, _):
+            if event['channel'] == await slack.channels.get(name='general').id:
+                await slack.channels.get('general', update=True)
+                members = await slack.channels.get(name='general').members
+                if (len(members) % 1000) == 0:
+                    to = await slack.channels.get(name='general').members
```
Comment (ovv) [inline sirbot/pythondev/slack.py]: `to` should be the channel not the members of the channel

**Review by ovv — CHANGES_REQUESTED** [review]
Some small changes but otherwise :+1:

File: sirbot/pythondev/slack.py (hunk 2 above)
Comment (nik849) [reply sirbot/pythondev/slack.py]: Ah yes... whoops!  I was hoping I could print our what sort of thing the slack.channels.get returned, but I couldn't test it - I think .members returns a list of members.  I didn't know I could just sent to the channel directly.  Ok, thanks for this - I'll make a few changes and update it, thanks @ovv!

**Review by ovv — APPROVED** [review]

**Pull request merged**